sql/ddl_internal.sql

```diff
@@ -47,38 +45,34 @@ BEGIN
         END IF;
     END IF;
 
-    FOR chunk_row IN SELECT *
-        FROM _timescaledb_catalog.chunk c
-        INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = c.hypertable_id)
-        INNER JOIN _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
-        INNER JOIN _timescaledb_catalog.dimension_slice ds
-            ON (ds.dimension_id = time_dimension.id)
-        INNER JOIN _timescaledb_catalog.chunk_constraint cc
-            ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
-        WHERE (older_than_time IS NULL OR ds.range_end <= older_than_time)
-        AND (drop_chunks_impl.schema_name IS NULL OR h.schema_name = drop_chunks_impl.schema_name)
-        AND (drop_chunks_impl.table_name IS NULL OR h.table_name = drop_chunks_impl.table_name)
+    FOR schema_name, table_name IN SELECT hyper.schema_name, hyper.table_name FROM _timescaledb_catalog.hypertable hyper WHERE
+                                        (drop_chunks_impl.schema_name IS NULL OR hyper.schema_name = drop_chunks_impl.schema_name) AND
+                                        (drop_chunks_impl.table_name IS NULL OR hyper.table_name = drop_chunks_impl.table_name)
     LOOP
-        IF truncate_before THEN
+        FOR chunk_row IN SELECT @extschema@.show_chunks(schema_name || '.' || table_name, older_than_time, newer_than_time)
+        LOOP
+            IF truncate_before THEN
+                EXECUTE format(
+                    $$
+                    TRUNCATE %s %s
+                    $$, chunk_row, cascade_mod
+                );
+            END IF;
             EXECUTE format(
-                $$
-                TRUNCATE %I.%I %s
-                $$, chunk_row.schema_name, chunk_row.table_name, cascade_mod
+                    $$
+                    DROP TABLE %s %s
+                    $$, chunk_row, cascade_mod
             );
-        END IF;
-        EXECUTE format(
-                $$
-                DROP TABLE %I.%I %s
-                $$, chunk_row.schema_name, chunk_row.table_name, cascade_mod
-        );
+        END LOOP;
     END LOOP;
 END
 $BODY$;
 
-CREATE OR REPLACE FUNCTION _timescaledb_internal.drop_chunks_type_check(
+CREATE OR REPLACE FUNCTION _timescaledb_internal.time_dim_type_check(
```

Comment:
Wait, this isn't called in any of your new chunk functions. Is there some future use planned?

Reply:
`export_chunk` will probably call it too.
Also, there is nothing specific to function of dropping chunks in the definition of this function so in any case I think it is better not to call it drop_chunks_type_check.